server/schema/customer/res.js

```diff
@@ -119,20 +119,24 @@ const resolvers = {
       return await Customer.create(input).then(customer => customer)
     },
     async applyGiftCard(_, {input}, {loggedInUser}) {
-      _auth(loggedInUser)
-      let {giftCard, expired} = await authGiftCard(input.jwt)
-      if (expired) throw new Error('This gift card has expired!')
-      if (giftCard.customerId) throw new Error('The gift card is not available anymore!')
-      let user = await Customer.findById(input.customerId)
-      if (user) {
-        let amount = giftCard.amount
-        let balance = user.get('balance') + amount
-        return await user.update({balance}).then(async () => {
-          return await giftCard.update({customerId: input.customerId}).then(() => {
-            return {balance, amount}
+      try {
+        _auth(loggedInUser)
+        let {giftCard, expired} = await authGiftCard(input.jwt)
+        if (expired) throw new Error('This gift card has expired!')
+        if (giftCard.customerId) throw new Error('The gift card is not available anymore!')
+        let user = await Customer.findById(input.customerId)
+        if (user) {
+          let amount = giftCard.amount
+          let balance = user.get('balance') + amount
+          return await user.update({balance}).then(async () => {
+            return await giftCard.update({customerId: input.customerId}).then(() => {
+              return {balance, amount}
+            })
           })
-        })
-      } else throw new Error('Not found Customer info!')
+        } else throw new Error('Not found Customer info!')
+      } catch (error) {
+        throw new Error(error)
```

Comment:
-> throw new Error(e.message)
becareful error here could be an Error Object, need to be sure how thrown error is processed on client side (action), would it accept both Error Obj and string msg -> so fallback to e.message could be a safer bet